collections-lists/ConvertIAsyncEnumerableToList/ConvertIAsyncEnumerableToListLibrary/UserPageFetcher.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ConvertIAsyncEnumerableToListLibrary.Model;
+using System.Runtime.CompilerServices;
+
+namespace ConvertIAsyncEnumerableToListLibrary;
+public static class UserPageFetcher
+{
+    public static async IAsyncEnumerable<User> GetUsersAsync(int delayMilliseconds = 500)
+    {
+        for (int page = 1; page <= 3; page++)
+        {
+            await Task.Delay(delayMilliseconds);
+
+            var users = new List<User>
+            {
+                new User((page - 1) * 3 + 1, "Alice"),
+                new User((page - 1) * 3 + 2, "Bob"),
+                new User((page - 1) * 3 + 3, "John")
+            };
+
+            foreach (var user in users)
+            {
+                yield return user;
+            }
+        }
+    }
+
+    public static async IAsyncEnumerable<User> GetUsersAsyncWithCancellationToken
+    (int delayMilliseconds = 500, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        for (int page = 1; page <= 3; page++)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await Task.Delay(delayMilliseconds, cancellationToken);
+
+            var users = new List<User>
+            {
+                new User((page - 1) * 3 + 1, "Alice"),
+                new User((page - 1) * 3 + 2, "Bob"),
+                new User((page - 1) * 3 + 3, "John")
+            };
+
+            foreach (var user in users)
+            {
+                yield return user;
```

Comment:
In the article, underneath where this snippet is located, you have this text:

> In this modified version, we pass the `CancellationToken` to `Task.Delay` to allow the delay to be cancelable. Additionally, we call `cancellationToken.ThrowIfCancellationRequested()` before yielding each user, as well as before the delay to ensure that the operation can be canceled at any point.

It looks like you meant to call `cancellationToken.ThrowIfCancellationRequested()` before yielding, but forgot to add that part to the code.